docs/examples-rust-wasi.md

```diff
@@ -1,101 +1,61 @@
 
 `wasi.rs`
 ```rust
 {{#include ../examples/wasi/wasm/wasi.rs}}
 ```
 
-Building this program generates `target/wasm32-wasip1/debug/wasi.wasm`, used below.
+> Building instructions:
+> 1. Have Rust installed
+> 2. Add WASIp2 target if you haven't already: `rustup target add wasm32-wasip2`
+> 3. `cargo build --target wasm32-wasip2`
 
-### Invoke the WASM module
+Building this program generates `target/wasm32-wasip2/debug/wasi.wasm`, used below.
 
-This example shows adding and configuring the WASI imports to invoke the above WASM module.
+### Invoke the WASM component
+
+This example shows adding and configuring the WASI imports to invoke the above WASM component.
 
 `main.rs`
 ```rust,ignore
 {{#include ../examples/wasi/main.rs}}
 ```
 
-## WASI state with other custom host state
-
-The [`add_to_linker`] takes a second argument which is a closure to access `&mut
-WasiCtx` from within the `T` stored in the `Store<T>` itself. In the above
-example this is trivial because the `T` in `Store<T>` is `WasiCtx` itself, but
-you can also store other state in `Store` like so:
-
-[`add_to_linker`]: https://docs.rs/wasi-common/*/wasi_common/sync/fn.add_to_linker.html
-[`Store`]: https://docs.rs/wasmtime/*/wasmtime/struct.Store.html
-[`BorrowMut<WasiCtx>`]: https://doc.rust-lang.org/stable/std/borrow/trait.BorrowMut.html
-[`WasiCtx`]: https://docs.rs/wasi-common/*/wasi_common/struct.WasiCtx.html
-
-```rust
-# extern crate wasmtime;
-# extern crate wasi_common;
-# extern crate anyhow;
-use anyhow::Result;
-use std::borrow::{Borrow, BorrowMut};
-use wasmtime::*;
-use wasi_common::{WasiCtx, sync::WasiCtxBuilder};
-
-struct MyState {
-    message: String,
-    wasi: WasiCtx,
-}
-
-fn main() -> Result<()> {
-    let engine = Engine::default();
-    let mut linker = Linker::new(&engine);
-    wasi_common::sync::add_to_linker(&mut linker, |state: &mut MyState| &mut state.wasi)?;
-
-    let wasi = WasiCtxBuilder::new()
-        .inherit_stdio()
-        .inherit_args()?
-        .build();
-    let mut store = Store::new(&engine, MyState {
-        message: format!("hello!"),
-        wasi,
-    });
-
-    // ...
-
-# let _linker: Linker<MyState> = linker;
-    Ok(())
-}
-```
-
-## WASI Preview 2
-
-An experimental implementation of the WASI Preview 2 API is also available, along with an adapter layer for  WASI Preview 1 WebAssembly modules. In future this `preview2` API will become the default. There are some features which are currently only accessible through the `preview2` API such as async support and overriding the clock and random implementations.
-
 ### Async example
 
-This [async example code][code2] shows how to use the [wasmtime-wasi::preview2][`preview2`] module to
-execute the same WASI Preview 1 WebAssembly module from the example above. This example requires the `wasmtime` crate `async` feature to be enabled.
+This [async example code][code2] shows how to use the [wasmtime-wasi][`wasmtime-wasi`] component to
```

Comment:
```suggestion
This [async example code][code2] shows how to use the [wasmtime-wasi][`wasmtime-wasi`] crate to
```

Here we're talking about a Rust crate rather than a Wasm component.

Reply:
yeah, that's a typo due to replacing "module" to "component"